compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitKeepScreenOnManager.uikit.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import platform.UIKit.UIApplication
+
+internal class UIKitKeepScreenOnManager private constructor(): PlatformKeepScreenOnManager {
+
+    val isKeepScreenOnEnabled: Boolean get() = _isKeepScreenOnEnabled
+
+    private var _isKeepScreenOnEnabled: Boolean
+        get() = UIApplication.sharedApplication.idleTimerDisabled
+        set(value) { UIApplication.sharedApplication.idleTimerDisabled = value }
+
+    private var keepScreenOnCount: Int = 0
+
+    override fun incrementKeepScreenOnCount() {
+        keepScreenOnCount++
+        updateIsKeepScreenOnEnabled()
+    }
+
+    override fun decrementKeepScreenOnCount() {
+        keepScreenOnCount--
+        updateIsKeepScreenOnEnabled()
+    }
+
+    private fun updateIsKeepScreenOnEnabled() {
+        _isKeepScreenOnEnabled = keepScreenOnCount > 0
+    }
+
+    /**
+     * Resets the keep screen on counter to zero and updates the idle timer state.
+     *
+     * This method is intended for testing purposes only.
+     */
+    fun reset() {
```

Comment:
Since `UIKitKeepScreenOnManager` is a singleton we either need to be able to mock it for tests or reset its state. To be able to mock it from the instrumented tests we would have to be able to pass the dependency all the way from the [`internal ComposeHostingViewController`](https://github.com/JetBrains/compose-multiplatform-core/blob/9d75c744b17d7fcd56f104ff80638078309fd5c5/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeHostingViewController.uikit.kt#L95) to `PlatformContext` but I am not sure if we want that so I have added this `reset` function. Is there some API that would make this function only available from tests? Or is there another way to approach this?